data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -152,26 +152,35 @@ private ParsedMessage convertS3EventMessages(final Message message) {
     }
 
     private List<DeleteMessageBatchRequestEntry> processS3EventNotificationRecords(final Collection<ParsedMessage> s3EventNotificationRecords) {
-        List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
+        final List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
+        final List<ParsedMessage> parsedMessagesToRead = new ArrayList<>();
```

Comment:
Worst case, is it fair to say this change doubles the space complexity of this method?

Reply:
No, I think not.

Space complexity: The `ParsedMessage` here is just a pointer, so we are perhaps increasing the number of pointers, but that is all. Each list contains a maximum of 10 items, so this is maybe 80 bytes.

Time complexity: These are two sequential loops. The code in the second loop was already happening in the first loop. It has just been moved out. So it is possibly O(2*N)  which is O(N). But since the code was just moved, I think that 2*N is not quite accurate either.

Reply:
Thanks, for clarifying. I forgot we aren't copying objects here and I agree its probably not exactly 2*N with the code shift.